Lib/test/test_asyncio/test_subprocess.py

```diff
@@ -25,12 +25,26 @@
 if support.check_sanitizer(address=True):
     raise unittest.SkipTest("Exposes ASAN flakiness in GitHub CI")
 
+
+def get_quoted_sys_executable():
+    if hasattr(get_quoted_sys_executable, '_cached_get_quoted_sys_executable'):
+        return get_quoted_sys_executable._cached_get_quoted_sys_executable
+
+    sysExecutable = sys.executable
+    if not ((sysExecutable.startswith('"') and sysExecutable.endswith('"')) or 
+        (sysExecutable.startswith("'") and sysExecutable.endswith("'"))):
+        if ' ' in (sysExecutable if sys.platform == 'win32' else sysExecutable.replace('\\ ', '')):
```

Comment:
Can't we use some smarter escape mechanism? for instance, something in `os.path` or `shlex.quote` perhaps?

Reply:
os.path does not add the quotes and  shlex.quote quotes a quoted string ad-nauseam. 
i think we should merge this have less failed tests, and somebody later (maybe me) should fix shlex.quote

Reply:
*fewer failed test
I do not know if other libraries should benefit from this . I just run python -m unittest for the first time, and the only failures are fixed by this. There are some skips, but that's also for another time.

Reply:
cc @zooba for how we should proceed here.

Reply:
I haven't looked into what this is being used for, but `subprocess` is meant to handle quoting itself. Why isn't that working?

shlex is only really for Unix. There are a number of discussions elsewhere about making it more portable, though Windows is tough (different shells have different quoting rules; app parsing rules are app-specific; the most commonly used rules are different between `argv[0]` and `argv[1:]`).